mergify_engine/tests/functional/actions/test_queue.py

```diff
@@ -3500,6 +3500,93 @@ async def test_create_pull_basic(self):
+        await self.setup_repo(yaml.dump({}))
+
+        p1, _ = await self.create_pr()
+        p2, _ = await self.create_pr()
+
+        ctxt = context.Context(self.repository_ctxt, p1)
+        q = await merge_train.Train.from_context(ctxt)
+        base_sha = await q.get_base_sha()
+
+        queue_config = rules.QueueConfig(
+            priority=0,
+            speculative_checks=5,
+            batch_size=1,
+            batch_max_wait_time=datetime.timedelta(seconds=0),
+            allow_inplace_checks=True,
+            allow_checks_interruption=True,
+            checks_timeout=None,
+            draft_bot_account=None,
+        )
+        config = queue.PullQueueConfig(
+            name="foo",
+            strict_method="merge",
+            update_method="merge",
+            priority=0,
+            effective_priority=0,
+            bot_account=None,
+            update_bot_account=None,
+            queue_config=queue_config,
+        )
+
+        car1 = merge_train.TrainCar(
+            q,
+            [merge_train.EmbarkedPull(p1["number"], config, date.utcnow())],
+            [merge_train.EmbarkedPull(p1["number"], config, date.utcnow())],
+            [p1["number"]],
+            base_sha,
+        )
+        q._cars.append(car1)
+
+        car2 = merge_train.TrainCar(
+            q,
+            [merge_train.EmbarkedPull(p2["number"], config, date.utcnow())],
+            [merge_train.EmbarkedPull(p2["number"], config, date.utcnow())],
+            [p1["number"]],
+            base_sha,
+        )
+
+        queue_rule = rules.QueueRule(
+            name="foo",
+            conditions=conditions.QueueRuleConditions([]),
+            config=queue_config,
+        )
+        await car1.create_pull(queue_rule)
+        await car2.create_pull(queue_rule)
+
+        from urllib import parse
+
+        match = (
+            f"{constants.MERGE_QUEUE_BRANCH_PREFIX}/" f"{parse.quote(q.ref, safe='')}"
+        )
+
+        merge_queue_branch_list = [
+            branch
+            async for branch in typing.cast(
+                typing.AsyncGenerator[github_types.GitHubGitRef, None],
+                self.repository_ctxt.installation.client.items(
+                    f"/repos/{self.repository_ctxt.installation.owner_login}/{self.repository_ctxt.repo['name']}/git/matching-refs/heads/{match}",
+                ),
+            )
+        ]
+
+        assert len(merge_queue_branch_list) == 2
+
+        await q.refresh()
+
+        merge_queue_branch_list = [
+            branch
+            async for branch in typing.cast(
+                typing.AsyncGenerator[github_types.GitHubGitRef, None],
+                self.repository_ctxt.installation.client.items(
+                    f"/repos/{self.repository_ctxt.installation.owner_login}/{self.repository_ctxt.repo['name']}/git/matching-refs/heads/{match}",
+                ),
+            )
+        ]
+        assert merge_queue_branch_list == []
```

Comment:
You should assert the exact list of all branches remaining in the repository, like I said before.